### ISSUE TITLE
Recommended Settings for 4K on Shield TV

### ISSUE THREAD

**Issue by JonRStorey** [koying/SPMC#1104]
Would it be useful to update the recommended 4k settings page to include a note for optional auto-resolution of video - where the shield can be set to 4k but SPMC GUI set to 1080p?

Using high quality Lanczos3 optimised scaler for sub 1080p media, this obviously then lets the TV upscale to 4k which seems to greatly improve the picture quality of 1080p (and below) media on my samsung 4k TV

**Comment by classicjazz** [comment]
Not sure I understand the request. 

If you have a UHD TV, you could chose to limit Shield resolution to 1080p to force your UHD TV to upscale. Of course, then Netflix/YouTube/Amazon 4K content would play at 1080p resolution but also be upscaled by the TV.

The Shield is powerful enough to run a 4K GUI on SPMC. So, if you use the default setting for the Shield at 4K resolution, then I recommend running both SPMC video and GUI at 4K too.

**Comment by JonRStorey** [comment]
Sorry classicjazz I was trying to say set the shield to 4k (so Netflix, amazon etc runs at 4k) but limit the video output resolution in SPMC to 1080p.
This will then play 4k and 1080p at their native resolutions and force anything below 1080p to be upscaled (using HQ lanczos3 optimised) to 1080p by the shield for then finally upscaled to 4k by the TV.
Noticeable improvement to PQ on my samsung UHD on anything 1080p and below while still retaining 4k output.

**Comment by iatacs19** [comment]
This is good info, can you update the wiki with the detailed settings? I would like to try it on my Sony 4K TV with my Shield TV. Right now I just have everything set to 4K.

**Comment by JonRStorey** [comment]
I just use all the standard settings as per the 4k setting page but in SPMC changed
system>video output>Resolution to 1920x1080p and refresh rate to 59.94 with GUI set to unlimited.

I have Mediacodec (Surface) turned off so that the HQ scaler is available for 720p video as i dont bother with HDR files in SPMC.

If that’s of concern you can just set another profile up with surface enabled just for your HDR media if required

**Comment by Alamei** [comment]
Just as a note for anyone reading this thread considering the same issue: setting the GUI resolution down is no longer necessary as of SPMC 17.6a2. It will now shift resolution up or down as necessary to match the file, regardless of GUI settings.

**Comment by JonRStorey** [comment]
Thanks Alamei i’ll check it out - so its auto-resolution without any settings change?

**Comment by Alamei** [comment]
Correct. It will attempt to adjust based on refresh rate and resolutions available, giving more weight to the former than the latter (e.g., if you're trying to play a 1080p25 file and the available resolutions are 2160p50 and 1080p24, you'll get 2160p50, since it's a "cleaner" match to the framerate).

**Comment by JonRStorey** [comment]
But to ensure correct resolution i’d still be better setting the GUI to 1080p then?

**Comment by Alamei** [comment]
I haven't run tests on what would happen if your GUI resolution already matched the target resolution but the framerate didn't and a matching framerate wasn't available at the current resolution. You'd have to ask @koying what would occur in that instance (or run a battery of tests yourself).

I would guess that the new switching code runs the same regardless of what your original resolution is (i.e., you'll end up with the same final res/framerate combo regardless of what res/framerate you start with), but I don't have any explicit knowledge to that point.
